pkg/controller/jaeger/secret_test.go

```diff
@@ -0,0 +1,131 @@
+package jaeger
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/strategy"
+)
+
+func TestSecretsCreate(t *testing.T) {
+	// prepare
+	nsn := types.NamespacedName{
+		Name: "TestSecretsCreate",
+	}
+
+	objs := []runtime.Object{
+		v1alpha1.NewJaeger(nsn.Name),
+	}
+
+	req := reconcile.Request{
+		NamespacedName: nsn,
+	}
+
+	r, cl := getReconciler(objs)
+	r.strategyChooser = func(jaeger *v1alpha1.Jaeger) strategy.S {
+		s := strategy.New().WithSecrets([]v1.Secret{{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nsn.Name,
+			},
+		}})
+		return s
+	}
+
+	// test
+	res, err := r.Reconcile(req)
+
+	// verify
+	assert.NoError(t, err)
+	assert.False(t, res.Requeue, "We don't requeue for now")
+
+	persisted := &v1.Secret{}
+	persistedName := types.NamespacedName{
+		Name:      nsn.Name,
+		Namespace: nsn.Namespace,
+	}
+	err = cl.Get(context.Background(), persistedName, persisted)
+	assert.Equal(t, persistedName.Name, persisted.Name)
+	assert.NoError(t, err)
+}
+
+func TestSecretsUpdate(t *testing.T) {
+	// prepare
+	nsn := types.NamespacedName{
+		Name: "TestSecretsUpdate",
+	}
+
+	orig := v1.Secret{}
+	orig.Name = nsn.Name
+	orig.Annotations = map[string]string{"key": "value"}
+
+	objs := []runtime.Object{
+		v1alpha1.NewJaeger(nsn.Name),
+		&orig,
+	}
+
+	r, cl := getReconciler(objs)
+	r.strategyChooser = func(jaeger *v1alpha1.Jaeger) strategy.S {
+		updated := v1.Secret{}
+		updated.Name = orig.Name
+		updated.Annotations = map[string]string{"key": "new-value"}
+
+		s := strategy.New().WithSecrets([]v1.Secret{updated})
+		return s
+	}
+
+	// test
+	_, err := r.Reconcile(reconcile.Request{NamespacedName: nsn})
+	assert.NoError(t, err)
+
+	// verify
+	persisted := &v1.Secret{}
+	persistedName := types.NamespacedName{
+		Name:      orig.Name,
+		Namespace: orig.Namespace,
+	}
+	err = cl.Get(context.Background(), persistedName, persisted)
+	assert.Equal(t, "new-value", persisted.Annotations["key"])
+	assert.NoError(t, err)
+}
+
+func TestSecretsDelete(t *testing.T) {
```

Comment:
The tests in this package look very similar. Possibly to dry them out?

Reply:
Same problem as with the main code: either we add complexity by making the code generic, or we accept the duplication and keep the code simple/flexible.